crates/cargo-contract/src/workspace/manifest.rs

```diff
@@ -289,7 +289,7 @@ impl Manifest {
         let ink_dylint = {
             let mut map = value::Table::new();
             map.insert("git".into(), "https://github.com/paritytech/ink/".into());
-            map.insert("branch".into(), "master".into());
+            map.insert("tag".into(), "v4.0.0-alpha.3".into());
```

Comment:
What this needs to do is to read out the version of the used ink! version. Otherwise linting a v3 ink contract will also disable linting.

Reply:
Ok, then I will need to do several more things to make this backwards compatible for v3 `ink` contracts. Alternatively we could say that you have to use `cargo-contract` `1.x` for `v3` contracts

Reply:
Yeah I think this is fine. We are still changing too much to be backwards compatible.

Reply:
https://github.com/paritytech/cargo-contract/issues/754